cmd/root.go

```diff
@@ -78,18 +78,17 @@ func (i *uint16Value) Type() string {
 func (i *uint16Value) String() string { return strconv.FormatUint(uint64(*i), 10) }
 
 func Execute() error {
-	egrp := errgroup.Group{}
+	egrp, egrpCtx := errgroup.WithContext(context.Background())
 	defer func() {
 		err := egrp.Wait()
 		if err != nil {
-			if err.Error() == "Federation process has been cancelled" {
-				log.Info("Process was shutdown")
-				return
-			}
-			log.Errorln("Error occurred when shutting down process:", err)
+			log.Errorln("Fatal error occured that leads to shutdown of the process:", err)
+		} else {
+			// Use Error instead of Info because our default log level is Error
+			log.Error("Pelican is safely exited")
 		}
 	}()
-	ctx := context.WithValue(context.Background(), config.EgrpKey, &egrp)
+	ctx := context.WithValue(egrpCtx, config.EgrpKey, egrp)
 	return rootCmd.ExecuteContext(ctx)
```

Comment:
This is a related bug (not caused by your PR but made it obvious).

If `serveCache` fails with an error (that is, the cache fails to startup), then nothing cancels the context, causing the process to hang indefinitely.

Contrast this to the `serveOrigin` process which wraps everything with a non-nil-error check:

```
func serveOrigin(cmd *cobra.Command, args []string) error {
	cancel, err := launchers.LaunchModules(cmd.Context(), config.OriginType)
	if err != nil {
		cancel()
	}

	return err
}
```

We either want to wrap `cacheServe` similarly or hoist the cancelation all the way up to here.